File: Week_01/G20190343020227/climbing-stairs.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n <= 3:return n
        arr = [0 for x in range(0,n+1)]
        arr[1],arr[2],index = 1,2,3
        while(index <= n):
            arr[index] = arr[index-1] + arr[index-2]
            index += 1
        return arr[n]

File: Week_05/G20190343010449/ClimbingStairs.java
class Solution {
    public int climbStairs(int n) {
        if (n <= 2) {
            return n;
        }
        return memo(n);
    }

    public int memo(int n) {
        int[] cache = new int[n + 1];
        cache[1] = 1;
        cache[2] = 2;

        for (int i = 3; i <= n; i++) {
            cache[i] = cache[i - 1] + cache[i - 2];
        }
        return cache[n];
    }
}